backup.mjs

```diff
@@ -0,0 +1,260 @@
+
+function run(cmd) {
+  return execSync(cmd).toString().trimEnd();
+}
+
+function getContexts() {
+  var stdout = run("kubectl config get-contexts -o name");
+  return stdout.split("\n");
+}
+
+// Sanity check
+
+var contexts = getContexts();
+if (!contexts.includes(stagingContext)) {
+  console.warn("Staging context not found. Tried", stagingContext, "but did not find it in", contexts);
+  console.warn("Might need to edit the top level of this file and try again");
+  process.exit(1);
+}
+if (!contexts.includes(prodContext)) {
+  console.warn("Prod context not found. Tried", prodContext, "but did not find it in", contexts);
+  console.warn("Might need to edit the top level of this file and try again");
+  process.exit(1);
+}
+
+// Process args
+
+if (process.argv.length < 3) {
+  console.warn("Please pass project ID or URL as argument, e.g. node backup.mjs 5dbf805650b51914727e06c4");
+  process.exit(2);
+}
+
+let projId;
+const arg = process.argv[2];
+if (URL.canParse(arg)) {
+  const url = new URL(arg);
+  if (url.pathname.startsWith("/app/lexicon/")) {
+    projId = url.pathname.substring("/app/lexicon/".length);
+  } else {
+    projId = url.pathname; // Will probably fail, but worth a try
+  }
+} else {
+  projId = arg;
+}
+
+projId = projId.trim();
+console.log("Project ID:", projId);
+
+// Start running
+
+console.warn("Setting up kubectl port forwarding for remote Mongo...");
+const remoteMongoPort = await randomFreePort();
+// TODO: Improve by finding a local port that's not in use, rather than hardcoding this
+let portForwardingReady;
+const portForwardingPromise = new Promise((resolve) => {
+  portForwardingReady = resolve;
+});
+portForwardProcess = spawn("kubectl", [`--context=${context}`, "port-forward", "svc/db", `${remoteMongoPort}:27017`], {
+  stdio: "pipe",
+});
+portForwardProcess.stdout.on("data", (data) => {
+  portForwardingReady();
+});
+portForwardProcess.stderr.on("data", (data) => {
+  console.warn("Port forwarding failed:");
+  console.warn(data.toString());
+  console.warn("Exiting");
+  process.exit(1);
+});
+
+console.warn("Setting up local Mongo connection...");
+
+const localMongoPort = run("docker compose port db 27017").split(":")[1];
+const localConnStr = `mongodb://admin:pass@localhost:${localMongoPort}/?authSource=admin`;
+localConn = await MongoClient.connect(localConnStr);
+
+const localAdmin = await localConn.db("scriptureforge").collection("users").findOne({ username: "admin" });
+const adminId = localAdmin._id.toString();
+console.log(`Local admin ID: ${adminId}`);
+console.warn("If that doesn't look right, hit Ctrl+C NOW");
+
+await portForwardingPromise;
+console.warn("Port forwarding is ready. Setting up remote Mongo connection...");
+
+const remoteConnStr = `mongodb://localhost:${remoteMongoPort}`;
+remoteConn = await MongoClient.connect(remoteConnStr);
+
+const remoteAdmin = await remoteConn.db("scriptureforge").collection("users").findOne({ username: "admin" });
```

Comment:
```suggestion
```
Probably remove this since it's not used?

Reply:
Done in commit 6d9bd5561.